docs/rfcs/0006-top-sites-feature.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: page
+title: Introduce a Top Sites Feature
+permalink: /rfc/0006-top-sites-feature
+---
+
+* Start date: 2020-07-30
+* RFC PR: [#7931](https://github.com/mozilla-mobile/android-components/pull/7931)
+
+## Summary
+
+Create a TopSitesFeature that abstracts out the co-ordination between the multiple data storages.
+
+## Motivation
```

Comment:
About the RFC process and not this PR: I wonder if it'd be good to add a section to collect a list of "requirements". For example, I made some suggestions about performance needs and telemetry we might want. As we collect feedback and iterate, how can we make sure we're continuing to meet all of these requirements if they're not documented? I'm concerned if they become documented in prose or in the solution, we'll forget about a few and future iterations will omit them.

Also, when looking through the RFC for historical purposes, it can be helpful to understand the requirements explicitly because, if we want to make a change to the API or replace the API, we can evaluate whether the same requirements apply or not.

CC @pocmo @csadilek